hathor/manager.py

```diff
@@ -486,7 +486,8 @@ def _initialize_components(self) -> None:
                         assert self.tx_storage.indexes is not None
                         if self.tx_storage.indexes.mempool_tips:
                             self.tx_storage.indexes.mempool_tips.update(tx)
-                    self.tx_storage.add_to_indexes(tx)
+                    if tx_meta.validation.is_fully_connected():
```

Comment:
No, we don't

Reply:
Should we let the index decide about adding it or not?

Reply:
I think that makes sense. The index manager already has access to what type of transaction each index wants, so this can work.

Reply:
Actually, because this isn't a direct call to the index-manager, I prefer to hold off on making this change now, mainly because of how it interacts with `_all_tips_cache`, which I think should move into the index manager. I'll leave a comment explaining that this should be refactored in the future.

Reply:
Now we have tests covering the initialization with a non-empty database. So I'm resolving this conversation.